### DISCUSSION TITLE
How to convert snake_case to camelCase in API response?

### DISCUSSION THREAD

**Discussion by kaisiant** [adonisjs/core#3161] [Help] [answered]
#1283 this answer is outdated.

Is it possible to convert the api response from snake_case to camelCase in a quick way?

I don't get the point to convert the camelCase to snake_case. Most frontends are dealing with camelCase right? Also Adonis model uses the camelCase as well. Only the database table column should be use snake_case.
![image](https://user-images.githubusercontent.com/26655111/132992256-e1527edc-77bd-4284-b765-07edca619cdc.png)

The output by default is already snake_case, I think adonis can provide another function like `serialize` to convert output to camelCase
```
public async index({ auth, request, response }: HttpContextContract) {
    const { page = 1, limit = 10, subcategorySlug } = request.qs()

    const productsQuery = Product.query()

    if (subcategorySlug) {
      productsQuery
        .join('sub_categories', 'sub_categories.id', 'products.subcategory_id')
        .where('sub_categories.slug', subcategorySlug)
        .select('products.*')
    }

    productsQuery.preload('images').preload('user').preload('subcategory')

    if (request.headers().site === 'Seller') {
      const currentUser = await auth.authenticate()
      productsQuery.where('products.user_id', currentUser.id)
    }

    const products = await productsQuery.paginate(page, limit)
    return products
  }
```

**Comment by kaisiant** [comment, marked as answer]
Problem solved by using this library in my frontend axios interceptor.
https://github.com/sindresorhus/camelcase-keys

**Comment by pk4all** [comment]
Use helpers
https://docs.adonisjs.com/guides/helpers#camelcase

**Reply by kaisiant** [reply]
I want to convert the whole api response to camelCase not a string
